src/server/pfs/db/driver.go

```diff
@@ -1165,12 +1167,9 @@ func getPrefixes(path string) []string {
 }
 
 func getDiffID(commitID string, path string) string {
-	return fmt.Sprintf("%s:%s", commitID, path)
-}
-
-// the equivalent of above except that commitID is a rethink term
-func getDiffIDFromTerm(commitID gorethink.Term, path string) gorethink.Term {
-	return commitID.Add(":" + path)
+	s := fmt.Sprintf("%s:%s", commitID, path)
+	hash := sha256.Sum256([]byte(s))
```

Comment:
I'm not sure how much work this is to change, but some advice I've gotten in the past is not to use cryptographic hashes if you don't need irreversibility, because they incur a performance hit.

Also, do pachyderm repos include a version number? If so, we can change our hashes later by providing clients with a pipeline that migrates data from an old-version repo to a new-version repo, but if not, we may at some point be stuck with whatever hash function we pick (because a customer has a bunch data in an existing pachyderm cluster, keyed by a particular hash function, and they don't want to lose it). I think it might be worth double-checking with JD how he feels about that risk.